Add logging to the front end

irisfaraway opened the pull request:
https://eaflood.atlassian.net/browse/WE-615

Add logging so we can understand what the app is doing, identify unexpected behaviour and diagnose issues.

Still a WIP right now.

irisfaraway commented:
As [Good is only intended to log Hapi events](https://github.com/hapijs/good/issues/308), we are leaving our `console.log`s and `console.error`s in their current format for now.

Requests for static assets no longer appear in the log though.

duncannixon commented on latestCommit.json:
```diff
@@ -0,0 +1 @@
+f7d2e924bdff99886feddd529c922b6b99285ec0
```
This file no longer exists as we are now saving the latest commit to an environment variable during the gulp task instead.

irisfaraway commented:
@Cruikshanks @duncannixon Think I've addressed all the issues from the review – would you mind having another look? PR looks massive now but most of the files have very minor changes only.

irisfaraway commented:
@Cruikshanks 😭  All good points. I'll sort this out today.

irisfaraway commented:
@Cruikshanks Think I've covered your comments now – would you mind having another look?
